### PR TITLE
Render cfg(feature) requirements in documentation

### PR DESCRIPTION
**Pull request by Nemo157**
Was based on #35903 when I implemented this almost a year ago, that's been closed since in favour of #1998, but since this is very much a special case for `cfg(feature)` I don't think it makes sense to update the issue id.

You can see the current output for diesel at https://nemo157.com/rustdoc-features-example/diesel-impl2/diesel/index.html, there is definitely some work on the HTML/CSS that needs to happen (e.g. [the implementors section has some really bad alignment issues](https://nemo157.com/rustdoc-features-example/diesel-impl2/diesel/types/trait.ToSql.html#implementors) but getting some feedback on the general structure + ideas about nicer ways to render the details into the docs would be great.

### CODE REVIEW

**Comment by GuillaumeGomez** [comment]
```
tidy error: /checkout/src/librustdoc/html/render.rs:2632: line longer than 100 chars
```

I'll take a look at the code when the tests pass.

File: src/librustdoc/html/static/rustdoc.css [hunk 1]
```diff
@@ -598,6 +598,19 @@ body.blur > :not(#help) {
 	top: 0;
 }
 
+.cfg-feature {
+	padding-left: 10px;
+	font-family: "Comic Sans MS", monospace;
```
Comment (retep998) [inline src/librustdoc/html/static/rustdoc.css]: `Comic Sans MS`
welp.

**Comment by Nemo157** [comment]
Sorry about that, pushed a fix

**Comment by kennytm** [comment]
Is it possible to do this via `#[doc(cfg)]` instead? For example make `#[cfg(feature = "x")]` imply `#[doc(cfg(feature = "x"))]`, and let the existing `doc(cfg)` code handle the rest (the rendering style would be different though).

**Comment by Nemo157** [comment]
@kennytm that would definitely help me avoid having to work out a nice rendering style :) I might try and see how that looks over the weekend

**Comment by carols10cents** [comment]
Friendly ping to keep this on your radar @Nemo157!

**Comment by alexcrichton** [comment]
I'm going to close this now due to inactivity to help clear out the queue, but please feel free to resubmit with a rebase!

**Pull request closed without merging**